docs: Extend sandbox management guide (#364)
daytonaio/docs: src/content/docs/sandbox-management.mdx

---

import { Tabs, TabItem } from '@astrojs/starlight/components';
import Image from 'astro/components/Image.astro';
import sandboxDiagram from '@assets/docs/sandbox-states.svg?raw';

Sandboxes are isolated development environments managed by Daytona. This guide covers how to create, manage, and remove Sandboxes using the SDK.
By default, Sandboxes auto-stop after 15 minutes of inactivity and use **1 vCPU**, **1GB RAM**, and **3GiB disk**.

## Sandbox Lifecycle

> A Daytona Sandbox can have three states during its lifetime: [Started](#creating-sandboxes), [Stopped](#stop-and-start-sandbox) and [Archived](#archive-sandbox). The diagram below shows the states and possible transitions between them.

Reviewer: ```suggestion
Throughout its lifecycle, a Daytona Sandbox can have several different states. The diagram below shows the states and possible transitions between them.
```

<Fragment set:html={sandboxDiagram} />

By default, sandboxes auto-stop after `15 minutes` of inactivity and auto-archive after `7 days` of being stopped.

## Creating Sandboxes

The Daytona SDK provides an option to create Sandboxes with default or custom configurations. You can specify the language, [Snapshot](/docs/snapshots), resources, environment variables, and volumes for the Sandbox.
Running Sandboxes utilize CPU, memory, and disk storage. Every resource is charged per second of usage.

:::tip

> If you want to prolong the auto-stop interval, you can [set the auto_stop_interval parameter](/docs/sandbox-management#auto-stop-interval) when creating a Sandbox.

Reviewer: ```suggestion
If you want to prolong the auto-stop interval, you can [set the auto-stop interval parameter](/docs/sandbox-management#auto-stop-interval) when creating a Sandbox.
```

// The sandbox ID can later be used to find the sandbox and start it

const sandbox = await daytona.findOne("7cd11133-96c1-4cc8-9baa-c757b8f8c916");

// Start Sandbox
await sandbox.start();
```

</TabItem>
</Tabs>

The stopped state should be used when the Sandbox is expected to be started again soon. Otherwise, it is recommended to stop and then archive the Sandbox to eliminate disk usage costs.

## Archive Sandbox

The Daytona SDK provides methods to archive Sandboxes using Python and TypeScript.

When Sandboxes are archived, the entire filesystem state is moved to very cost-effective object storage, making it possible to keep Sandboxes available for an extended period.
Starting an archived Sandbox takes more time than starting a stopped Sandbox, depending on its size.

A Sandbox must be stopped before it can be archived, and can be started again in the same way as a stopped Sandbox.

<Tabs>
<TabItem label="Python" icon="seti:python">
```python
# Archive Sandbox
sandbox.archive()
```
</TabItem>
<TabItem label="TypeScript" icon="seti:typescript">
```typescript
// Archive Sandbox
await sandbox.archive();
```
</TabItem>
</Tabs>

> ## Remove Sandbox

Reviewer: ```suggestion
## Delete Sandbox
```

To get the preview URL for a specific port, check out [Preview & Authentication](/docs/preview-and-authentication).

## Stop and Resume Sandbox

> The Daytona SDK provides methods to stop and resume Sandboxes using Python and TypeScript.

Reviewer 2: Use `start` instead of `resume`.

We have plans on implementing actual `pause` and `resume` methods that will pause/resume the memory state as well so we don't want that to conflict with `start`.

### Auto-archive Interval

> The `auto_archive_interval` parameter sets the amount of time after which a stopped Sandbox will be automatically archived.

Reviewer: ```suggestion
The auto-archive interval parameter sets the amount of time after which a continuously stopped Sandbox will be automatically archived.
```
